components/automate-ui/src/app/modules/license/license-apply/license-apply.component.spec.ts

```diff
@@ -40,7 +40,7 @@ function genLicenseResp(licenseEndDate: moment.Moment): LicenseStatus {
 }
 
 describe('LicenseApplyComponent', () => {
-  let component: LicenseApplyComponent;
+  let component;
```

Comment:
Please revert this--should always use specific types rather than declaring variables as type `any`, which is what this does.

Same change in license-lockout.component.spec.ts, too.

Reply:
actually `licenseFacade` is private and `let component: LicenseApplyComponent;` is not allowed to call directly, if I use `let component: LicenseApplyComponent;` then I am getting
`Property 'licenseFacade' is private and only accessible within class 'LicenseLockoutComponent'.ts(2341)
`

Reply:
Ah, so that means the code was actually hiding that quite valid error message--you should not be able to access `licenseFacade` in the test. I understand why you made the change--ideally one should not have to modify the code-under-test just for the benefit of the test.
However, it is more important not to hide any valid errors--a somewhat better solution in this case is to make `licenseFacade` public. (Yes, it is a not-ideal change to make, but it is the lesser of two evils.)

Reply:
Yes, I had this solution before, but I didn't change it.

Reply:
I have updated the changes.